ca/server.go

```diff
@@ -530,13 +534,8 @@ func (s *Server) evaluateAndSignNodeCert(ctx context.Context, node *api.Node) {
 
 // signNodeCert does the bulk of the work for signing a certificate
 func (s *Server) signNodeCert(ctx context.Context, node *api.Node) {
-	if !s.securityConfig.RootCA().CanSign() {
-		log.G(ctx).WithFields(logrus.Fields{
-			"node.id": node.ID,
-			"method":  "(*Server).signNodeCert",
-		}).Errorf("no valid signer found")
-		return
-	}
+	rootCA := s.securityConfig.RootCA()
+	externalCA := s.securityConfig.ExternalCA()
```

Comment:
Oh, I think I found it: `RenewTLSConfig()` in `ca/config.go` appears to be the only place where `(* MutableTLSCreds ).LoadNewTLSConfig()` is called. I will add update method to the `*ExternalCA` type so that you can give it a new TLS config and list of URLs. This will allow us to only create the ExternalCA once and store it in the `SecurytConfig` then we can create a new HTTP client when the TLS config changes and update the list of URLs whenever there is a cluster spec change. This will also allow HTTP connection reuse while the TLS config has not changed.